MDANSE/Src/MDANSE/Framework/Configurators/WeightsConfigurator.py

```diff
@@ -114,18 +114,21 @@ def get_weights(self):
         ]
 
         weights = {}
-        for i in range(atom_selection_configurator["selection_length"]):
-            name = atom_selection_configurator["names"][i]
-            for element in atom_selection_configurator["elements"][i]:
+        for name, elements in itertools.islice(
+            zip(
+                atom_selection_configurator["names"],
+                atom_selection_configurator["elements"],
+            ),
+            atom_selection_configurator["selection_length"],
+        ):
+            for element in elements:
                 property = self._trajectory.get_atom_property(element, self["property"])
                 if name in weights:
                     weights[name] += property
                 else:
                     weights[name] = property
```

Comment:
This form makes it more obvious that what might be happening here is:
```suggestion
            weights[name] = sum(self._trajectory.get_atom_property(element, self["property"])
                                for element in elements)
```

Reply:
I've added the change but needed to update it slightly see [43d4118](https://github.com/ISISNeutronMuon/MDANSE/pull/619/commits/43d41186c31e556519b217a85170fa44840bc32b).